builder/assignmentEnvImageBuilder.go

```diff
@@ -19,36 +21,41 @@ import (
 	"strings"
 )
 
-type baseEnvImage interface {
-	verifyAndWriteInstructions() error
-	build() error
-	publish() error
-	resetDockerfileData()
-	deleteDockerfile() error
-	undoBuild() error
-}
-
-type assignmentEnvironment struct {
+// assignmentEnvironmentImageBuilder is used to build an image with an environment
+// that satisfies all the language and library dependencies of an assignment.
+//
+// The image is built in phases to allow for rollback in situations where the execution of a phase fails.
+// Once the configuration passes a verification phase, a Dockerfile is created and
+// the docker client (here - https://github.com/moby/moby/tree/master/client) is used to generate the image.
+// The image is then pushed to docker hub repository, if required.
+type assignmentEnvironmentImageBuilder struct {
 	DockerfileInstructions bytes.Buffer
 	ImgBuildConfig         *imageBuildConfig
 	AssgnEnvConfig         *configurations.AssignmentEnvConfig
 	ImageExists            bool
 }
 
-type assignmentEnvironmentOption func(*assignmentEnvironment) error
+// assignmentEnvironmentOption is a function interface that is supplied as different
```

Comment:
is function interface a thing?
Maybe rewrite the comment as shown below. You'll need to first rename the option to
`assignmentEnvironmentImageBuilderOption`.
```go
// assignmentEnvironmentImageBuilderOption represents options that can be used to help initialize
// an instance of assignmentEnvironmentImageBuilder.
// Each option is a closure that is responsible for initializing one or more members
// while instantiating assignmentEnvironmentImageBuilder.
```

Reply:
Okay.